nack/responder_option.go

```diff
@@ -0,0 +1,21 @@
+package nack
+
+import "github.com/pion/logging"
+
+// ResponderOption can be used to configure ResponderInterceptor
+type ResponderOption func(s *ResponderInterceptor)
```

Comment:
I think it's better to make it
```suggestion
type ResponderOption func(s *ResponderInterceptor) error
```
and validate the arguments. (`ResponderSize` should be 2^n)

Reply:
I think it would be better to do this only for the size option, eg:

// ResponderSize sets the size of the interceptor.
// Size must be one of: 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 32768
func ResponderSize(size uint16) (ResponderOption, error) {
	
	_, err := newSendBuffer(r.size)
	if err != nil {
		return nil, err
	}

	return func(r *ResponderInterceptor) {
		r.size = size
	}, nil
}

This way we can also remove the error return from NewResponderInterceptor
same applies to GeneratorInterceptor

Reply:
Fixed! Both options now return errors. That could come in handy in the future also!